llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -797,12 +806,22 @@ processInputModule(std::unique_ptr<Module> M) {
     SplitByOptionalFeatures |= OptionalFeaturesSplitter->remainingSplits() > 1;
 
     while (OptionalFeaturesSplitter->hasMoreSplits()) {
-      TopLevelModules.emplace_back(OptionalFeaturesSplitter->nextSplit());
+      // Here, we perform third-level splitting based on optimization level.
```

Comment:
It would be great if we can get this checked in in the current shape and take an action item to improve the code within an acceptable amount of time. It will be beneficial to have this feature out sooner than later.